File: A/579/579.py
input_value = int(input())
count = 1

while input_value > 1:
    if input_value % 2 != 0:
        count += 1
        input_value -= 1
    else:
        input_value //= 2   # divisão inteira, equivalente ao /= 2 em C++
        
print(count)

File: A/579/579.cpp
#include <iostream>
#include <cmath>

int main() {
	int input;
	std::cin >> input;
	int count = 1;
	while (input > 1) {
		if (input % 2 != 0) {
			count++;
			input--;
		}
		else {
			input /= 2;
		}
	}
	std::cout << count;
}
